<<<<<<< HEAD
var apply = require('./internal/apply'),
    isError = require('./isError'),
    rest = require('./rest');
=======
var apply = require('./_apply'),
    baseRest = require('./_baseRest'),
    isError = require('./isError');
>>>>>>> origin/master

/**
 * Attempts to invoke `func`, returning either the result or the caught error
 * object. Any additional arguments are provided to `func` when it's invoked.
 *
 * @static
 * @memberOf _
<<<<<<< HEAD
 * @category Util
 * @param {Function} func The function to attempt.
 * @returns {*} Returns the `func` result or error object.
 * @example
 *
 * // avoid throwing errors for invalid selectors
=======
 * @since 3.0.0
 * @category Util
 * @param {Function} func The function to attempt.
 * @param {...*} [args] The arguments to invoke `func` with.
 * @returns {*} Returns the `func` result or error object.
 * @example
 *
 * // Avoid throwing errors for invalid selectors.
>>>>>>> origin/master
 * var elements = _.attempt(function(selector) {
 *   return document.querySelectorAll(selector);
 * }, '>_>');
 *
 * if (_.isError(elements)) {
 *   elements = [];
 * }
 */
<<<<<<< HEAD
var attempt = rest(function(func, args) {
=======
var attempt = baseRest(function(func, args) {
>>>>>>> origin/master
  try {
    return apply(func, undefined, args);
  } catch (e) {
    return isError(e) ? e : new Error(e);
  }
});

module.exports = attempt;
